tests/e2e/parallel/main_test.go

```diff
@@ -2290,6 +2291,1146 @@ func TestScheduledSuiteUpdate(t *testing.T) {
+			Title:       "Custom Security Checks",
+			Description: "Test profile using CEL-based CustomRules",
+			EnableRules: []compv1alpha1.RuleReferenceSpec{
+				{
+					Name:      customRuleName,
+					Kind:      "CustomRule",
+					Rationale: "Security best practice requires pods to run as non-root",
+				},
+			},
+		},
+	}
+
+	err = f.Client.Create(context.TODO(), tp, nil)
+	if err != nil {
+		t.Fatalf("Failed to create TailoredProfile: %v", err)
+	}
+	defer f.Client.Delete(context.TODO(), tp)
+
+	// Create ScanSettingBinding
+	ssb := &compv1alpha1.ScanSettingBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ssbName,
+			Namespace: testNamespace,
+		},
+		Profiles: []compv1alpha1.NamedObjectReference{
+			{
+				APIGroup: "compliance.openshift.io/v1alpha1",
+				Kind:     "TailoredProfile",
+				Name:     tpName,
+			},
+		},
+		SettingsRef: &compv1alpha1.NamedObjectReference{
+			APIGroup: "compliance.openshift.io/v1alpha1",
+			Kind:     "ScanSetting",
+			Name:     "default",
+		},
+	}
+
+	err = f.Client.Create(context.TODO(), ssb, nil)
+	if err != nil {
+		t.Fatalf("Failed to create ScanSettingBinding: %v", err)
+	}
+	defer f.Client.Delete(context.TODO(), ssb)
+
+	// Wait for suite to be created and for scans to complete
+	suiteName := ssbName
+
+	// Wait for scans to complete
+	// The scan should be NON-COMPLIANT because our test pod doesn't have the required security context
+	err = f.WaitForSuiteScansStatus(testNamespace, suiteName, compv1alpha1.PhaseDone, compv1alpha1.ResultCompliant)
+	if err != nil {
+		t.Fatalf("Scan did not complete as expected: %v", err)
+	}
+	// Create a test pod without security context (should fail the check)
+	testPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-test-pod", testName),
+			Namespace: testNamespace,
+			Labels: map[string]string{
+				"customrule-test": testLabel,
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    "test-container",
+					Image:   "busybox:latest",
+					Command: []string{"sh", "-c", "sleep 3600"},
+				},
+			},
+			// Deliberately not setting securityContext to test the CustomRule
+		},
+	}
+
+	// Create test pod
+	err = f.Client.Create(context.TODO(), testPod, nil)
+	if err != nil {
+		t.Fatalf("Failed to create test pod: %v", err)
+	}
+	defer f.Client.Delete(context.TODO(), testPod)
+
+	suite := &compv1alpha1.ComplianceSuite{}
+	key := types.NamespacedName{Name: suiteName, Namespace: testNamespace}
+	if err := f.Client.Get(context.TODO(), key, suite); err != nil {
+		t.Fatal(err)
+	}
+	// let's rescans and expect the check to be compliant by deleting the suite
+	err = f.Client.Delete(context.TODO(), suite)
+	if err != nil {
+		t.Fatalf("Failed to delete suite: %v", err)
+	}
+	err = f.WaitForSuiteScansStatus(testNamespace, suiteName, compv1alpha1.PhaseDone, compv1alpha1.ResultNonCompliant)
+	if err != nil {
+		t.Fatalf("Scan did not complete as expected: %v", err)
+	}
```

Comment:
Could add some additional validation here to check that the check result for that particular rule is "FAIL".